wgsl/index.bs

```diff
@@ -11737,8 +10999,861 @@ When calling a built-in function, all arguments to the function are evaluated
+      <td>`T` is a [=type/concrete=] [=scalar=]<br>
+      `S` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a three-component [=vector=] with `e.x`, `e.y`, and `e.z` as components.
+
+      If `T` does not match `S` a conversion is used and the components are `T(e.x)`, `T(e.y)`, and  `T(e.z)`.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec3<T>(e1 : T, e2 : T, e3 : T) -> vec3<T>
+@const @must_use fn vec3(e1 : T, e2 : T, e3 : T) -> vec3<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a three-component [=vector=] with `e1`, `e2`, and `e3` as components.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec3<T>(v1 : vec2<T>, e1 : T) -> vec3<T>
+@const @must_use fn vec3(v1 : vec2<T>, e1 : T) -> vec3<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a three-component [=vector=] with `v1.x`, `v1.y`, and `e1` as components.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec3<T>(e1 : T, v1 : vec2<T>) -> vec3<T>
+@const @must_use fn vec3(e1 : T, v1 : vec2<T>) -> vec3<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a three-component [=vector=] with `e1`, `v1.x`, and `v1.y` as components.
+</table>
+
+#### `vec4` #### {#vec4-builtin}
+
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec4<T>(e : vec4<S>) -> vec4<T>
+@const @must_use fn vec4(e : vec4<S>) -> vec4<S>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is a [=type/concrete=] [=scalar=]<br>
+      `S` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a four-component [=vector=] with `e.x`, `e.y`, `e.z`, and `e.w` as components.
+
+      If `T` does not match `S` a conversion is used and the components are `T(e.x)`, `T(e.y)`, `T(e.z)` and  `T(e.w)`.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec4<T>(e1 : T, e2 : T, e3 : T, e4 : T) -> vec4<T>
+@const @must_use fn vec4(e1 : T, e2 : T, e3 : T, e4 : T) -> vec4<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a four-component [=vector=] with `e1`, `e2`, `e3`, and `e4` as components.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec4<T>(e1 : T, v1 : vec2<T>, e2 : T) -> vec4<T>
+@const @must_use fn vec4(v1 : vec2<T>, v2 : vec2<T>) -> vec4<T>
```

Comment:
What you have here as the second overload is already present two pairs below.

I think this has to be vec4(e1:T, v1:vec2<T>, e2:T)